src/events/attributes.rs

```diff
@@ -130,6 +149,50 @@ impl<'a> Attribute<'a> {
         String::from_utf8(unescaped.into_owned()).map_err(|e| Error::Utf8(e.utf8_error()))
     }
 
+    /// Decode then unescapes the value with custom entities
+    ///
+    /// This allocates a `String` in all cases. For performance reasons it might be a better idea to
+    /// instead use one of:
+    ///
+    /// * [`Reader::decode()`], as it only allocates when the decoding can't be performed otherwise.
+    /// * [`unescaped_value()`], as it doesn't allocate when no escape sequences are used.
+    ///
+    /// [`unescaped_value_with_custom_entities()`]: #method.unescaped_value
+    /// [`Reader::decode()`]: ../../reader/struct.Reader.html#method.decode
+    #[cfg(feature = "encoding")]
+    pub fn unescape_and_decode_value_with_custom_entities<B: BufRead>(
+        &self,
+        reader: &Reader<B>,
+        custom_entities: &HashMap<Vec<u8>, Vec<u8>>,
+    ) -> Result<String> {
+        let decoded = reader.decode(&*self.value);
+        let unescaped =
+            unescape_with(decoded.as_bytes(), custom_entities).map_err(Error::EscapeError)?;
+        String::from_utf8(unescaped.into_owned()).map_err(|e| Error::Utf8(e.utf8_error()))
```

Comment:
Are we sure it'll be utf8? Shouldn't we use the `reader.decode` instead?

Reply:
I lean towards keeping things simple: what about requiring that `custom_entities` contains UTF-8?

Reply:
About making the code more DRY: yes, I would be better, but I wanted to test the idea with you before refactoring... I'll go make the change (as well as your comments above).

Reply:
Alright!